File: src/Web/src/model/Device/DeviceDto.ts
import { Vendor } from "../../enum/Vendor";
import { AeroMetadata } from "./AeroMetadata";
import { AmicoMetadata } from "./AmicoMetadata";
import { DeviceModuleDto } from "./DeviceModuleDto";

export interface DeviceDto {
  guid: string;
  name: string;
  serialNumber: string;
  mac: string;
  ip: string;
  port: string;
  firmware: string;
  vendor: Vendor;
  metadata: AeroMetadata | AmicoMetadata | string;
  syncedAt: Date;
  configurationStatus: string;
  deviceModules: DeviceModuleDto[];
  locationGuid: string;
  isDefault: boolean;
  isActive: boolean;
}

File: src/Modules/Contracts/Core.Contract/DTOs/Device/DeviceDto.cs
using SharedKernel.Enums;

namespace Core.Contract.DTOs.Device;

public sealed record DeviceDto(
  Guid Guid,
  string Name,
  string SerialNumber,
  string Mac,
  string Ip,
  int Port,
  string Firmware,
  Vendor Vendor,
  string Metadata,
  DateTime SyncedAt,
  string ConfigurationStatus,
  Guid LocationGuid,
  bool IsActive,
  bool IsDefault
);